src/test/java/org/apache/sysds/test/functions/federated/algorithms/FederatedLmPipeline.java

```diff
@@ -76,37 +85,48 @@ public void federatedLmPipeline(ExecMode execMode, boolean contSplits) {
 			X = rc.append(X, new MatrixBlock(), true);
 			
 			// We have two matrices handled by a single federated worker
-			int halfRows = rows / 2;
-			writeInputMatrixWithMTD("X1", X.slice(0, halfRows-1), false);
-			writeInputMatrixWithMTD("X2", X.slice(halfRows, rows-1), false);
+			int quarterRows = TEST_NAME.equals(TEST_NAME2) ? rows / 4 : rows / 2;
+			List<Integer> k =  TEST_NAME.equals(TEST_NAME2) ? List.of(quarterRows-1, quarterRows, 2*quarterRows-1, 2*quarterRows, 3*quarterRows-1, 3*quarterRows, rows-1) :
+				List.of(quarterRows-1, quarterRows, rows-1, 0, 0, 0, 0);
+			writeInputMatrixWithMTD("X1", X.slice(0, k.get(0)), false);
+			writeInputMatrixWithMTD("X2", X.slice(k.get(1), k.get(2)), false);
+			writeInputMatrixWithMTD("X3", X.slice(k.get(3), k.get(4)), false);
+			writeInputMatrixWithMTD("X4", X.slice(k.get(5), k.get(6)), false);
 			writeInputMatrixWithMTD("Y", y, false);
 			
 			// empty script name because we don't execute any script, just start the worker
 			fullDMLScriptName = "";
 			int port1 = getRandomAvailablePort();
 			int port2 = getRandomAvailablePort();
+			int port3 = getRandomAvailablePort();
+			int port4 = getRandomAvailablePort();
 			Thread t1 = startLocalFedWorkerThread(port1, FED_WORKER_WAIT_S);
 			Thread t2 = startLocalFedWorkerThread(port2);
+			Thread t3 = startLocalFedWorkerThread(port3);
```

Comment:
I have added the second argument to reduce the Federated worker startup, so we don't have to wait for a second for each of the workers, but just the last.